ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala

```diff
@@ -454,6 +497,8 @@ object ContractsService {
         override def findByContractId(
             ctx: SearchContext[Option, Option],
             contractId: domain.ContractId,
+        )(implicit
+            lc: LoggingContextOf[CorrelationID with RequestID]
```

Comment:
Something to think about when declaring types of your `lc` implicit parameters: is there something essential about `findByContractId` that says "this must be called in the scope of a request"? If not, the intersection should not mention `RequestID`, and the variable should be `LoggingContextOf[Any]` in the extreme. A counterexample would be the functions in `Endpoints`, for which I _would_ intrinsically expect a `RequestID` to be present.

In general, I would expect that the deeper into the callstack you are, the wider the type parameter of `LoggingContextOf` is; that is the pattern that implicit resolution supports without you having to do anything special. So at the outset of a request you are more likely to see `LoggingContextOf[CorrelationID with RequestID]`, and ultimate utility-only functions take only `LoggingContextOf[Any]`, if anything.

Reply:
Triggers provide an example of this; `LoggingContextOf[Config with Trigger]` is requested by _trigger service_ methods, but it calls out to the lower-level trigger _runner_, which only requests `LoggingContextOf[Trigger]` (because service configuration is not a relevant concept to the runner).

Reply:
You're right about this. I didn't think about using Any yet although I already thought that it is not practical to enforce `CorrelationID with RequestID` everywhere.